core/domain/email_manager.py

```diff
@@ -376,16 +376,31 @@ def send_mail_to_admin(email_subject, email_body):
         body, body.replace('\n', '<br/>'), bcc_admin=False)
 
 
-def send_post_signup_email(user_id):
+def send_post_signup_email(user_id, test_for_duplicate_email=False):
     """Sends a post-signup email to the given user.
 
     Raises an exception if emails are not allowed to be sent to users (i.e.
     feconf.CAN_SEND_EMAILS is False).
 
     Args:
         user_id: str. User ID of the user that signed up.
+        test_for_duplicate_email: bool. For testing duplicate emails.
     """
 
+    # For testing the method _send_email.
+    if test_for_duplicate_email:
+        user_settings = user_services.get_user_settings(user_id)
+        email_subject = SIGNUP_EMAIL_CONTENT.value['subject']
+        email_body = 'Hi %s,<br><br>%s<br><br>%s' % (
+            user_settings.username,
+            SIGNUP_EMAIL_CONTENT.value['html_body'],
+            EMAIL_FOOTER.value)
+
+        _send_email(
+            user_id, feconf.SYSTEM_COMMITTER_ID, feconf.EMAIL_INTENT_SIGNUP,
+            email_subject, email_body, feconf.NOREPLY_EMAIL_ADDRESS)
+        return
+
```

Comment:
There's too much code duplication here. What you probably want is a not condition and not run lines 406-410